### ISSUE TITLE
Replacing the last characters of a line works incorrectly.

### ISSUE THREAD

**Issue by yixiang** [XVimProject/XVim#1083]
# Description
Select and replace last characters of a line works incorrectly. Instead of appending the register content after the last character, it inserts the characters before the last character.

# Operation
Say we have a line of code like this:
`int var;`
  1. Copy `int' so we have something in the register
  2. Select `ar;`
  3. Hit key 'p'.
  4. The result becomes `int intv`.

# Expected behaviour
The replacement result should be `int vint`.

**Comment by brianlheim** [comment]
I believe this is a duplicate of #1047 (but better title)

**Comment by brianlheim** [comment]
@lyxera yes that is true (see my comment on that issue). one of them should be closed.
